### PR TITLE
KEYCLOAK-17648 Make transaction timeout configurable

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.redhat.com/browse/KEYCLOAK-17648

### CODE REVIEW

File: modules/os-eap7-launch/added/launch/tx-datasource.sh [hunk 1]
```diff
@@ -85,6 +86,10 @@ function inject_jdbc_store() {
                 <communication table-prefix=\"${prefix}\"/>\\
                 <state table-prefix=\"${prefix}\"/>\\
             </jdbc-store>"
+  if [ -n "${2}" ]; then
+    jdbcStore="<coordinator-environment default-timeout=\"${2}\"></coordinator-environment>\\
+            $jdbcStore"
```
Comment (ASzc) [inline modules/os-eap7-launch/added/launch/tx-datasource.sh:91]: Doesn't `coordinator-environment` go in a different namespace than all the `jdbc-store`s? Schema [here](https://github.com/wildfly/wildfly/blob/master/transactions/src/main/resources/schema/wildfly-txn_5_0.xsd#L142) seems to suggest it should be under the transactions subsystem, whereas the stores are under the datasource subsystem?

If so, a simple concatenation at the beginning of the datastore list won't work

**Review by ASzc — CHANGES_REQUESTED** [review]
Seems like the code will do what it wants to do, except one detail of where exactly the timeout needs to go in the xml. See the inline comment. I've never configured the timeout before though, so I could be wrong.

File: modules/os-eap7-launch/added/launch/tx-datasource.sh (hunk 1 above)
Comment (slaskawi) [reply modules/os-eap7-launch/added/launch/tx-datasource.sh:91]: The naming here is less than perfect, but I think I put it in the right place.

The result of this function is put into `##JDBC_STORE##` tag (or marker) in the `standalone-openshift.xml`. You may find the skeleton for it [here](https://github.com/jboss-container-images/redhat-sso-7-openshift-image/blob/sso74-dev/modules/sso/config/launch/setup/74/added/standalone-openshift.xml#L534). As you probably noticed, the `##JDBC_STORE##` marker is placed under `<subsystem xmlns="urn:jboss:domain:transactions:5.0">`.

So I think maybe the `##JDBC_STORE##` should be called `##DETAILED_TRANSACTION_SETTINGS##` or something? But I don't want to introduce such change in this PR.

Comment (pskopek) [reply modules/os-eap7-launch/added/launch/tx-datasource.sh:91]: It looks good to me! Though the variable name is not really helpful here.

Comment (ASzc) [reply modules/os-eap7-launch/added/launch/tx-datasource.sh:91]: Ok, looks good then

**Review by ASzc — APPROVED** [review]

**Pull request merged**